### PR TITLE
das: add test for DASer

### PR DESCRIPTION
**Pull request by renaynay**
Adds a test for the `DASer`.

### CODE REVIEW

File: das/daser_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	sub := &mockHeaderSub{
+		headers: []*header.ExtendedHeader{randHeader},
+	}
+
+	daser := NewDASer(shareServ, sub)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		daser.sampling(context.Background(), sub)
```
Comment (Wondertan) [inline das/daser_test.go:28]: The reason I was really lazy about writing the test here is that I didn't want to make a mock Subscriber which returns a mock subscription. Your solution to the problem is great!

File: das/daser.go [hunk 2]
```diff
@@ -65,13 +65,11 @@ func (d *DASer) sampling(ctx context.Context, sub header.Subscription) {
 	for {
 		h, err := sub.NextHeader(ctx)
 		if err != nil {
-			if err == context.Canceled {
-				return
+			if err != context.Canceled {
+				log.Errorw("DASer failed to get next header", "err", err)
```
Comment (Wondertan) [inline das/daser.go:69]: Wait, wat? The code is identical before and after

File: das/daser.go [hunk 3]
```diff
@@ -65,13 +65,11 @@ func (d *DASer) sampling(ctx context.Context, sub header.Subscription) {
...
 			}
-			log.Errorw("DASer failed to get next header", "err", err)
 			return
 		}
-
```
Comment (Wondertan) [inline das/daser.go:74]: And this spacing is right?

File: das/daser.go (hunk 2 above)
Comment (Wondertan) [reply das/daser.go:69]: Ok, I see, it makes it one line shorter.

**Comment by Wondertan** [comment]
@renaynay, pls fix lint issues and we can merge it.

File: das/daser.go [hunk 4]
```diff
@@ -12,10 +12,10 @@ import (
 
 var log = logging.Logger("das")
 
+// DASer continuously validates availability of data committed to headers.
 // TODO(@Wondertan): Initialization and warm-up.
 // TODO(@Wondertan): Start and Stop is better be thread-safe.
 // TODO(@Wondertan): Add some testing
```
Comment (Wondertan) [inline das/daser.go:18]:
```suggestion
```

File: das/daser.go [hunk 5]
```diff
@@ -12,10 +12,10 @@ import (
 
 var log = logging.Logger("das")
 
+// DASer continuously validates availability of data committed to headers.
 // TODO(@Wondertan): Initialization and warm-up.
```
Comment (Wondertan) [inline das/daser.go:16]:
```suggestion
```

This is described in #181

**Review by Wondertan — APPROVED** [review]
LGTM

**Review by Wondertan — APPROVED** [review]

File: das/daser.go [hunk 6]
```diff
@@ -12,10 +12,8 @@ import (
 
 var log = logging.Logger("das")
 
-// TODO(@Wondertan): Initialization and warm-up.
-// TODO(@Wondertan): Start and Stop is better be thread-safe.
-// TODO(@Wondertan): Add some testing
 // DASer continuously validates availability of data committed to headers.
+// TODO(@Wondertan): Start and Stop is better be thread-safe.
 type DASer struct {
```
Comment (liamsi) [inline das/daser.go:17]: Orthogonal to this PR: The name DASer is always a bit confusing to me. Let's think of a better name long term.

**Review by liamsi — APPROVED** [review]
👏🏼

File: das/daser.go (hunk 6 above)
Comment (Wondertan) [reply das/daser.go:17]: @liamsi, totally agree. I was thinking about `da.Sync`, as essentially, it performs syncing of available shares. This component can be reused for FullSync of shares or Namespace sync, depending on shares.Availability implementation given to him.

Comment (liamsi) [reply das/daser.go:17]: Let's keep it for now. And change the name later.

**Pull request merged**